### DISCUSSION TITLE
adjusting the display-math margin

### DISCUSSION THREAD

**Discussion by nschloe** [KaTeX/KaTeX#3637] [Q&A]
When using KaTeX on GitHub, the top-margin on display math is too large:

---
![screenshot](https://user-images.githubusercontent.com/181628/169868584-3af1068f-7b24-4f7d-b535-f6fc528cc7fe.png)
---

That's because GitHub already applies a margin of 16 at the end of each paragraph. Reducing the top margin to 0 fixes things:

---
![screenshot](https://user-images.githubusercontent.com/181628/169868709-75a1aa6e-a654-43e2-8b22-171b5046b9ba.png)
---

Is this adjustable as KaTeX option?

**Comment by edemaine** [comment]
The margin is defined here:

https://github.com/KaTeX/KaTeX/blob/023cc0342cac5b2f3582f6de156b16d770437ebb/src/katex.less#L642

Can you override this CSS rule? I'm not sure exactly in what context you're using KaTeX -- GitHub Pages?

**Reply by nschloe** [reply]
Thanks Erik for the swift reply! I'm talking about [xhub](https://github.com/nschloe/xhub), the browser extension that loads KaTeX (and other things) onto GitHub pages.

I could potentially adapt that margin when packaging xhub. Do you know if that's possible programatically? Maybe even better I could add my own little less file to override just that one property.

**Reply by nschloe** [reply]
Could a variable like
```less
@display-margin: 1em 0;
...
margin: @display-margin;
```
be used in `katex.less`? That'd make adapting the margin easy for me.

**Reply by nschloe** [reply]
https://github.com/KaTeX/KaTeX/pull/3638

**Comment by ylemkimon** [comment]
The display margin rule is the least-specific rule (`.katex-display`), and you can override it with a more specific CSS rule.

**Reply by nschloe** [reply]
I have no power over the CSS in the page.
